jupyter_distutils.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+from glob import glob
+import logging as log
+import os
+from os.path import join as pjoin
+import pipes
+
+# BEFORE importing distutils, remove MANIFEST. distutils doesn't properly
+# update it when the contents of directories change.
+if os.path.exists('MANIFEST'): os.remove('MANIFEST')
+
+from distutils.cmd import Command
+from distutils.command.build_py import build_py
+from distutils.command.sdist import sdist
+from subprocess import check_call
+import sys
+
+try:
+    from wheel.bdist_wheel import bdist_wheel
+except ImportError:
+    bdist_wheel = None
+
+if any(arg.startswith('bdist') for arg in sys.argv):
```

Comment:
In this day and age, I think we should import setuptools unconditionally and just disable implicit bdist_egg, as we do [here](https://github.com/jupyterhub/jupyterhub/blob/0.7.2/setup.py#L283). Lots of things break nowadays when packages are installed without setuptools metadata.